var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -127,7 +134,9 @@ def install(self, spec, prefix):
                 'complex' if '+complex' in spec else 'real'),
             '--with-shared-libraries=%s' % ('1' if '+shared' in spec else '0'),
             '--with-debugging=%s' % ('1' if '+debug' in spec else '0'),
-            '--with-blas-lapack-dir=%s' % spec['lapack'].prefix
+            '--with-lapack-lib=%s' % spec['lapack'].lapack_static_lib,
```

Comment:
@davydden Another thing that you'll see failing when using MKL is that there is no single blas or lapack library. I think we should embrace python protocols and make all singular `*_lib` plurals and iterables. For `openblas` it would be something like : 
```
self.lapack_static_libs = ('libopenblas.so', )
```
but MKL will have a long list of libraries depending on the actual case at hand. What do you think ?